app/models/apple/publisher.rb

```diff
@@ -91,8 +91,11 @@ def deliver_and_publish!(eps)
             upload_media!(eps)
           end
 
-          process_and_deliver!(eps)
+          eps.filter(&:apple_needs_delivery?).tap do |eps|
```

Comment:
nice, good use of that status check, and makes sense to filter before calling `process_and_deliver!`